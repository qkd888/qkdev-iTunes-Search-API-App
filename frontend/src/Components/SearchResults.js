import React, { Component } from "react"; // This import statement allows access to the React library
import ResultsItem from "./ResultsItem"; // importing the ResultsItem component

export default class SearchResults extends Component {
  // Stateless component declaration
  render() {
    const { results, addToFavourites, clearResults } = this.props; // Destructuring props

    // If the results array has any items then
    // mapping through the array and returning
    // a ResultsItem component for each item.
    // The ResultsItem component takes in
    // item artistName, item trackName, item collectionName,
    // item artwork and the addFavourites function  as props

    if (results.length > 0) {
      return (
        <div className=" container resultsContainer">
          <h1 className=" heading mt-5 mb-5 ">Results</h1>
          {results.map((item, index) => {
            return (
              <div key={index}>
                <ResultsItem
                  artwork={item.artworkUrl100}
                  artistName={item.artistName}
                  trackName={item.trackName}
                  collectionName={item.collectionName}
                  addToFavourites={() => addToFavourites(item.trackId)}
                />
              </div>
            );
          })}
          {/*Clear results button which invokes the clearResults function when clicked */}
          <button className=" mt-5 mb-5 clearBtn" onClick={clearResults}>
            {" "}
            Clear Results <i className="fa fa-trash"></i>
          </button>{" "}
        </div>
      );
    } else {
      // Else return an H3 heading informing the user that there are no results
      return <h3 className="mt-5">No results</h3>;
    }
  }
}
