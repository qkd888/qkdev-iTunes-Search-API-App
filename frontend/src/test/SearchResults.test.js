import React from "react"; // This import statement allows access to the React library
import SearchResults from "../Components/SearchResults"; // importing the SearchResults component
import renderer from "react-test-renderer"; // importing the renderer  component from react-test-renderer

// Snapshot tests for the SearchResults component

// Testing if the SearchResults component renders properly if its passed an empty array
test("Renders correctly with empty array", () => {
  const tree = renderer.create(<SearchResults results={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing if the SearchResults component renders properly if its passed a populated array
test("Renders correctly with populuated array", () => {
  const results = [
    {
      trackId: 1,
      artistName: "j cole",
      trackName: "4 your eyes",
      collectionName: "4 your eyes only",
      artworkUrl100: "www.loremIp.com",
    },
  ];
  const tree = renderer.create(<SearchResults results={results} />).toJSON();
  expect(tree).toMatchSnapshot();
});
