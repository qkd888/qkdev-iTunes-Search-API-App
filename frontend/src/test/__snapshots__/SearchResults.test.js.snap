// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with empty array 1`] = `
<h3
  className="mt-5"
>
  No results
</h3>
`;

exports[`Renders correctly with populuated array 1`] = `
<div
  className=" container resultsContainer"
>
  <h1
    className=" heading mt-5 mb-5 "
  >
    Results
  </h1>
  <div>
    <div
      className=" shadow resultItem"
    >
      <div>
        <img
          alt="Thumbnail"
          height={100}
          src="www.loremIp.com"
          width={100}
        />
        <p
          className="artistName"
        >
          j cole
           
        </p>
        <p
          className="trackName"
        >
          4 your eyes
           
        </p>
        <p
          className="collectionName"
        >
          4 your eyes only
           
        </p>
        <button
          className="addFavourite"
          onClick={[Function]}
        >
          Add To Favourites
        </button>
      </div>
    </div>
  </div>
  <button
    className=" mt-5 mb-5 clearBtn"
  >
     
    Clear Results 
    <i
      className="fa fa-trash"
    />
  </button>
   
</div>
`;
